### PR TITLE
Update detection metrics to remove rdt

### LINKED ISSUE #170: Use `sklearn`'s OneHotEncoder instead of `rdt`
### Problem Description

Replace `rdt.transformers.OneHotEncoder` with `sklearn.preprocessing.OneHotEncoder` in the following metrics:

```
LogisticDetection,
SVCDetection,
LogisticParentChildDetection
SVCParentChildDetection
```

### Expected behavior

External metric behavior should not change.

### PR DESCRIPTION
**Pull request by katxiao**
Resolves #170

### CODE REVIEW

**Review by amontanez24 — APPROVED** [review]
Seems good to me. Nice work!

**Review by fealho — COMMENTED** [review]
👍

**Review by fealho — APPROVED** [review]
